reporters/ReportVectorMigration.cpp

```diff
@@ -166,33 +220,61 @@ GetReportInstantiator( Kernel::instantiator_function_t* pif )
             throw QueryInterfaceException(__FILE__, __LINE__, __FUNCTION__, "pvc", "IMigrate", "IVectorCohort");
         }
 
+        VectorStateEnum::Enum state = pvc->GetState();
+        if (!m_MustBeInState.empty())
+        {
+            if (std::find(m_MustBeInState.begin(), m_MustBeInState.end(), state) == m_MustBeInState.end())
+            {
+                // state is not one of the required states
+                return;
+            }
+        }
+
         uint32_t vci_id = pvc->GetID();
         int from_node_id = pSim->GetNodeExternalID( nodeSuid ) ;
+        if (!m_MustBeFromNode.empty())
+        {
+            if (std::find(m_MustBeFromNode.begin(), m_MustBeFromNode.end(), from_node_id) == m_MustBeFromNode.end())
+            {
+                // from_node_id is not one we want in the report
+                return;
+            }
+        }
+
         int to_node_id = pSim->GetNodeExternalID( pim->GetMigrationDestination() ) ;
-        int mig_type = pim->GetMigrationType() ;
-        VectorStateEnum::Enum state = pvc->GetState();
+        if (!m_MustBeToNode.empty())
+        {
+            if (std::find(m_MustBeToNode.begin(), m_MustBeToNode.end(), to_node_id) == m_MustBeToNode.end())
+            {
+                // to_node_id is not one we want in the report
+                return;
+            }
+        }
         std::string species = pvc->GetSpecies();
+        if (!m_SpeciesList.empty())
```

Comment:
I'd change:

std::string species = pvc->GetSpecies();

to

const std::string& species = pvc->GetSpecies(); 

so we don't copy the string.